File: Robot Mesh Studio/C++/V5 Clawbot Example.cpp
//LINK TO SOURCE: https://www.robotmesh.com/studio/5be40c6ec8f17a1f5795f745
// VEX V5 C++ Project
#include "vex.h"

//#region config_globals
vex::brain      Brain;
vex::motor      motor_arm(vex::PORT1, vex::gearSetting::ratio18_1, false);
vex::motor      motor_claw(vex::PORT10, vex::gearSetting::ratio18_1, false);
vex::motor      motor_right(vex::PORT15, vex::gearSetting::ratio18_1, true);
vex::motor      motor_left(vex::PORT16, vex::gearSetting::ratio18_1, false);
vex::controller con(vex::controllerType::primary);
//#endregion config_globals

using namespace vex;

int main(void) {
    //these control the velocity the motor will pick when not given a velocity
    motor_arm.setVelocity(35,velocityUnits::pct);
    motor_claw.setVelocity(35,velocityUnits::pct);
    while(true) {
        
        //arm
        if(con.ButtonL1.pressing()) {
            //raise arm
            motor_arm.spin(directionType::fwd);
        } else if (con.ButtonL2.pressing()) {
            //lower arm
            motor_arm.spin(directionType::rev);
        } else {
            //hold position
            motor_arm.stop(brakeType::hold);
        }
        
        //claw
        if(con.ButtonR1.pressing()) {
            //open claw
            motor_claw.spin(directionType::fwd);
        } else if (con.ButtonR2.pressing()) {
            //close claw
            motor_claw.spin(directionType::rev);
        } else {
            //hold position
            motor_claw.stop(brakeType::hold);
        }
        
        //right drive motor
        motor_right.spin(directionType::fwd,con.Axis2.value(),velocityUnits::pct);
        
        //left drive motor
        motor_left.spin(directionType::fwd,con.Axis3.value(),velocityUnits::pct);
    }
}

File: Robot Mesh Studio/Python/V5 Clawbot Example.py
#LINK TO SOURCE: https://www.robotmesh.com/studio/5be40c6ec8f17a1f5795f60d
import vex

#region config
brain       = vex.Brain();
motor_arm   = vex.Motor(vex.Ports.PORT1, vex.GearSetting.RATIO18_1, False)
motor_claw  = vex.Motor(vex.Ports.PORT10, vex.GearSetting.RATIO18_1, False)
motor_right = vex.Motor(vex.Ports.PORT15, vex.GearSetting.RATIO18_1, True)
motor_left  = vex.Motor(vex.Ports.PORT16, vex.GearSetting.RATIO18_1, False)
con         = vex.Controller(vex.ControllerType.PRIMARY)
#endregion config

while True:
  #arm
  if con.buttonL1.pressing():
    # Raise arm
    motor_arm.spin(vex.DirectionType.FWD, 35, vex.VelocityUnits.PCT)
  elif con.buttonL2.pressing():
    # Lower arm
    motor_arm.spin(vex.DirectionType.REV, 35, vex.VelocityUnits.PCT)
  else:
    # Arm hold still
    motor_arm.stop(vex.BrakeType.HOLD)
  #claw
  if con.buttonR1.pressing():
    # Open claw
    motor_claw.spin(vex.DirectionType.FWD, 35, vex.VelocityUnits.PCT)
  elif con.buttonR2.pressing():
    # Close claw
    motor_claw.spin(vex.DirectionType.REV, 35, vex.VelocityUnits.PCT)
  else:
    # Claw hold still
    motor_claw.stop(vex.BrakeType.HOLD)
  # Right stick to right motor
  motor_right.spin(vex.DirectionType.FWD, (con.axis2.position(vex.PercentUnits.PCT)), vex.VelocityUnits.PCT)
  # Left stick to left motor
  motor_left.spin(vex.DirectionType.FWD, (con.axis3.position(vex.PercentUnits.PCT)), vex.VelocityUnits.PCT)
